shared/common-adapters/usernames/index.tsx

```diff
@@ -152,62 +146,52 @@ UsernameText.defaultProps = {
-    const rwers = this.props.users.filter(u => !u.readOnly)
-    const readers = this.props.users.filter(u => !!u.readOnly)
-    const bgMode = this.props.backgroundMode || null
-    const isNegative = backgroundModeIsNegative(bgMode)
+const _Usernames = (props: Props) => {
+  const containerStyle = props.inline ? styles.inlineStyle : styles.nonInlineStyle
+  const rwers = props.users.filter(u => !u.readOnly)
+  const readers = props.users.filter(u => !!u.readOnly)
+  const bgMode = props.backgroundMode || null
+  const isNegative = backgroundModeIsNegative(bgMode)
 
-    return (
-      <Text
-        type={this.props.type}
-        negative={isNegative}
-        style={Styles.collapseStyles([containerStyle, this.props.containerStyle])}
-        title={this.props.title}
-        ellipsizeMode="tail"
-        lineClamp={this.props.lineClamp}
-        {...(this.props.inline ? inlineProps : {})}
-      >
-        {!!this.props.prefix && (
-          <Text type={this.props.type} negative={isNegative} style={this.props.style}>
-            {this.props.prefix}
-          </Text>
-        )}
-        <UsernameText {...this.props} users={rwers} />
-        {!!readers.length && (
-          <Text
-            type={this.props.type}
-            negative={isNegative}
-            style={Styles.collapseStyles([this.props.style, {marginRight: 1}])}
-          >
-            #
-          </Text>
-        )}
-        <UsernameText {...this.props} users={readers} />
-        {!!this.props.suffix && (
-          <Text
-            type={this.props.suffixType || this.props.type}
-            negative={isNegative}
-            style={Styles.collapseStyles([this.props.style, {marginLeft: Styles.globalMargins.xtiny}])}
-          >
-            {this.props.suffix}
-          </Text>
-        )}
-      </Text>
-    )
-  }
+  return (
+    <Text
+      type={props.type}
+      negative={isNegative}
+      style={Styles.collapseStyles([containerStyle, props.containerStyle])}
+      title={props.title}
+      ellipsizeMode="tail"
+      lineClamp={props.lineClamp}
+      {...(props.inline ? inlineProps : {})}
+    >
+      {!!props.prefix && (
+        <Text type={props.type} negative={isNegative} style={props.style}>
+          {props.prefix}
+        </Text>
+      )}
+      <UsernameText {...props} users={rwers} />
+      {!!readers.length && (
+        <Text
+          type={props.type}
+          negative={isNegative}
+          style={Styles.collapseStyles([props.style, {marginRight: 1}])}
+        >
+          #
+        </Text>
+      )}
+      <UsernameText {...props} users={readers} />
+      {!!props.suffix && (
+        <Text
+          type={props.suffixType || props.type}
+          negative={isNegative}
+          style={Styles.collapseStyles([props.style, {marginLeft: Styles.globalMargins.xtiny}])}
+        >
+          {props.suffix}
+        </Text>
+      )}
+    </Text>
+  )
 }
+const Usernames = React.memo(_Usernames)
```

Comment:
What does this do?

Reply:
https://reactjs.org/docs/react-api.html#reactmemo